mantaray/received.py

```diff
@@ -312,13 +327,27 @@ def _handle_cap(server_view: views.ServerView, args: list[str]) -> None:
     if subcommand == "ACK":
         acknowledged = set(args[-1].split())
         if "sasl" in acknowledged:
+            server_view.core.cap_list.add("sasl")
             server_view.core.send("AUTHENTICATE PLAIN")
+        if "away-notify" in acknowledged:
+            server_view.core.cap_list.add("away-notify")
     elif subcommand == "NAK":
         rejected = set(args[-1].split())
         if "sasl" in rejected:
             # TODO: this good?
             raise ValueError("The server does not support SASL.")
 
+    server_view.core.number_of_cap_req -= (
+        1  # To evaluate how many more ACK/NAKs will be received from server
```

Comment:
This comment IMO belongs to where `number_of_cap_req` is defined. I also don't like the name. How about `pending_capabilities` or `pending_cap_count`, pending as in we don't know yet whether server has it or not?